tests/sparseml/pytorch/optim/test_manager.py

```diff
@@ -41,6 +42,413 @@
+      params:
+      - re:.*weight
+      start_epoch: 0.0
+      update_frequency: 1.0
+stage_0:
+  metadata:
+    level: 1
+    metadata: null
+  stage_0_modifiers:
+  - !EpochRangeModifier
+      end_epoch: 6.0
+      start_epoch: 3.0
+  - !GMPruningModifier
+      end_epoch: 6.0
+      final_sparsity: 0.8
+      init_sparsity: 0.2
+      inter_func: cubic
+      leave_enabled: true
+      log_types: __ALL__
+      mask_type: unstructured
+      params:
+      - re:.*weight
+      start_epoch: 3.0
+      update_frequency: 1.0
+stage_1:
+  metadata:
+    level: 1
+    metadata: null
+  stage_1_modifiers:
+  - !ACDCPruningModifier
+      compression_sparsity: 0.5
+      end_epoch: 12
+      leave_enabled: true
+      log_types: __ALL__
+      mask_type: unstructured
+      momentum_buffer_reset: true
+      params:
+      - re:.*weight
+      start_epoch: 8
+      update_frequency: 2
+  - !EpochRangeModifier
+      end_epoch: 12.0
+      start_epoch: 6.0
+"""
+
+FOUR_STAGES_RECIPE = """version: 1.1.0
+
+stage_0:
+  metadata:
+    level: 0
+    metadata: null
+  stage_0_modifiers:
+  - !EpochRangeModifier
+      end_epoch: 3.0
+      start_epoch: 0.0
+  - !GMPruningModifier
+      end_epoch: 3.0
+      final_sparsity: 0.8
+      init_sparsity: 0.2
+      inter_func: cubic
+      leave_enabled: true
+      log_types: __ALL__
+      mask_type: unstructured
+      params:
+      - re:.*weight
+      start_epoch: 0.0
+      update_frequency: 1.0
+stage_1:
+  metadata:
+    level: 1
+    metadata: null
+  stage_1_modifiers:
+  - !ACDCPruningModifier
+      compression_sparsity: 0.5
+      end_epoch: 6
+      leave_enabled: true
+      log_types: __ALL__
+      mask_type: unstructured
+      momentum_buffer_reset: true
+      params:
+      - re:.*weight
+      start_epoch: 5
+      update_frequency: 2
+  - !EpochRangeModifier
+      end_epoch: 9.0
+      start_epoch: 3.0
+stage_3:
+  metadata:
+    level: 1
+    metadata: null
+  stage_3_modifiers:
+  - !EpochRangeModifier
+      end_epoch: 12.0
+      start_epoch: 9.0
+  - !GMPruningModifier
+      end_epoch: 12.0
+      final_sparsity: 0.8
+      init_sparsity: 0.2
+      inter_func: cubic
+      leave_enabled: true
+      log_types: __ALL__
+      mask_type: unstructured
+      params:
+      - re:.*weight
+      start_epoch: 9.0
+      update_frequency: 1.0
+stage_4:
+  metadata:
+    level: 1
+    metadata: null
+  stage_4_modifiers:
+  - !ACDCPruningModifier
+      compression_sparsity: 0.5
+      end_epoch: 16
+      leave_enabled: true
+      log_types: __ALL__
+      mask_type: unstructured
+      momentum_buffer_reset: true
+      params:
+      - re:.*weight
+      start_epoch: 14
+      update_frequency: 2
+  - !EpochRangeModifier
+      end_epoch: 18.0
+      start_epoch: 12.0
+"""
+METADATA = """{{'metadata': None, 'level': {level}}}"""
+
+# TODO: Investigate a peculiar bug, in `optim/modifier.py Line 319
+# yaml.safe_load() improperly parses end_epoch for ACDC
```

Comment:
Those tests fail due to solely one, weird behavior which only pertains to the 'end_epoch' attribute of ACDCModifier. I have found that the incorrect parsing in `optim/modifier` at L319 is the culprit. @bfineran let's take a look at it together,  remain clueless for now :)